File: frontend/src/Services/feedbackService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

// Get all feedback
const getAllFeedback = async () => {
  const response = await axios.get(`${API_URL}/feedback`);
  return response.data;
};

// Get a single feedback by ID
const getFeedbackById = async (id) => {
  const response = await axios.get(`${API_URL}/feedback/${id}`);
  return response.data;
};

// Create a new feedback
const createFeedback = async (feedbackData) => {
  const response = await axios.post(`${API_URL}/feedback`, feedbackData);
  return response.data;
};

// Update a feedback by ID
const updateFeedback = async (id, feedbackData) => {
  const response = await axios.put(`${API_URL}/feedback/${id}`, feedbackData);
  return response.data;
};

// Delete a feedback by ID
const deleteFeedback = async (id) => {
  const response = await axios.delete(`${API_URL}/feedback/${id}`);
  return response.data;
};

// Get feedback by email
const getFeedbackByEmail = async (email) => {
  const response = await axios.get(`${API_URL}/feedback/email/${email}`);
  return response.data;
};



const feedbackService = {
  getAllFeedback,
  getFeedbackById,
  createFeedback,
  updateFeedback,
  deleteFeedback,
  getFeedbackByEmail
};

export default feedbackService;

File: backend/src/main/java/com/stock/backend/service/FeedbackService.java
package com.stock.backend.service;

import com.stock.backend.model.Feedback;
import com.stock.backend.repository.FeedbackRepository;
import jakarta.persistence.EntityNotFoundException;
import jakarta.transaction.Transactional;
import lombok.extern.log4j.Log4j2;
import org.springframework.stereotype.Service;

import java.util.List;

@Log4j2
@Service
@Transactional
public class FeedbackService {

    private final FeedbackRepository feedbackRepository;

    public FeedbackService(FeedbackRepository feedbackRepository) {
        this.feedbackRepository = feedbackRepository;
    }

    public Feedback createFeedback(Feedback feedback) {
        return feedbackRepository.save(feedback);
    }

    public Feedback getFeedbackById(Integer id) {
        return feedbackRepository.findById(id)
                .orElseThrow(() -> new EntityNotFoundException("Feedback not found with id: " + id));
    }

    public Feedback getFeedbackByEmail(String email) {
        return feedbackRepository.findFeedbackByEmail(email)
                .orElseThrow(() -> new EntityNotFoundException("Feedback not found with Email: " + email));
    }

    public List<Feedback> getAllFeedback() {
        return feedbackRepository.findAll();
    }

    public Feedback updateFeedback(Integer id, Feedback feedback) {
        Feedback existingFeedback = getFeedbackById(id);
        existingFeedback.setName(feedback.getName());
        existingFeedback.setEmail(feedback.getEmail());
        existingFeedback.setSubject(feedback.getSubject());
        existingFeedback.setMessage(feedback.getMessage());
        existingFeedback.setStatus(feedback.getStatus());
        return feedbackRepository.save(existingFeedback);
    }

    public void deleteFeedback(Integer id) {
        Feedback feedback = getFeedbackById(id);
        feedbackRepository.delete(feedback);
    }

}
